src/utils/api.js

```diff
@@ -206,6 +212,25 @@ export const getSubjectsPromises = () => {
     .catch(error => ({ error }));
 };
 
+export const deleteSubjectPromise = (subjectId: SubjectId) => {
+  const token = getToken();
+  if (!token) {
+    throw new Error('Unauthorized action');
```

Comment:
Redirect to login maybe?

Should probably make the `fetch` code more general (used everywhere) so that we can redirect to login when the token is expired / get 401 from backend for all requests

Reply:
yeah that's a good point.

Created a ticket for it here https://github.com/veritus/veritus-web/issues/80

Not sure if this logic should live in `fetch` or Redux

Reply:
should probably do it in a middleware that does a request, success and error I think, like in #64

Reply:
Closed #80 in favour of #64